mmcv/device/ipu/hierarchical_data_manager.py

```diff
@@ -68,12 +70,11 @@ def hierarchical_data(self):
         return self._hierarchical_data
 
     def update_hierarchical_data(self,
-                                 dataA,
-                                 dataB=HierarchicalDataNone,
-                                 strict=True,
-                                 address='data'):
+                                 dataA: Any,
+                                 dataB: Any = HierarchicalDataNone,
```

Comment:
Why do the Type Hints of `dataA` and `dataB`  conflict with the docstring?

Reply:
Docstring cannot pass the detection of mypy，I tryed. After reading the source code, I found that these two parameters are not just the types in docstring

Reply:
> Why do the Type Hints of `dataA` and `dataB` conflict with the docstring?

Look that [lint](https://app.circleci.com/pipelines/github/open-mmlab/mmcv/712/workflows/15e2c001-a255-4828-b2d6-91ca354cf506/jobs/1411)

Reply:
It seems `dataB` should not be a tuple since its immutable property is conflicts with `dataB[idx] = node`. Will
 `dataB: Sequence` solve these conflicts?

Reply:
> It seems `dataB` should not be a tuple since its immutable property is conflicts with `dataB[idx] = node`. Will `dataB: Sequence` solve these conflicts?

I dont think so. because the code like this :
```       
       elif isinstance(dataA, self.atomic_types):
            if not self.quick_mode:
                is_equal = self.compare_atomic_type(dataA, dataB)
                if not is_equal:
                    if strict:
                        raise ValueError(
                            'all data except torch.Tensor should be same, '
                            f'but data({address}) is changed.')
                    else:
                        self.warning(
                            f'find a non-torch.Tensor data({type(dataA)}) '
                            f'changed, and the address is {address}')
        elif isinstance(dataA, DataContainer):
            if not self.quick_mode:
                assert isinstance(dataB, DataContainer)
                new_address = address + '.data'
                self.update_hierarchical_data(
                    dataA.data, dataB.data, False, address=new_address)```
 U can find this:  
```
        self.atomic_types = (int, str, float, np.ndarray, type(None))
```

 I think this means that the input types of dataA and dataB should be consistent, but the code does not make logical judgments.So I said he wasn't just these types, and int/str/ are  not Sequence, this is not rigorous.
I just aim at the current function. You can consider the overall situation.